### ISSUE TITLE
Add nnBench.rank command for statistical comparison of multiple configurations

### ISSUE THREAD

**Issue by aymuos15** [aymuos15/nnBenchmark#15]
## Feature Request

Add a new CLI command `nnBench.rank` to automatically perform statistical ranking and comparison across multiple configurations/models on the same dataset.

## Motivation

When comparing multiple model configurations (e.g., different architectures, hyperparameters, or loss functions) on the same dataset, researchers need to:
- Compare performance metrics across configurations
- Perform statistical significance testing
- Generate ranking tables for publications

Inspired by ranking methodologies used in medical imaging challenges (e.g., BraTS, KiTS), this feature would enable systematic comparison of different approaches.

## Proposed CLI Interface

```bash
# Compare multiple configurations on a dataset
nnBench.rank --dataset Dataset001_Example --configs fold_0.yaml fold_1.yaml custom_config.yaml

# Or use config names directly if they're in the standard results directory
nnBench.rank --dataset Dataset001_Example --configs fold_0 fold_1 custom_model
```

## Expected Output

The command should generate:

1. **Ranking Table** (`ranking_summary.csv`):
   - Rows: Different configurations
   - Columns: Metrics (Dice, Surface Dice, etc.) with mean ± std
   - Ranked by primary metric (e.g., mean Dice score)

2. **Statistical Tests** (`statistical_tests.json`):
   - Pairwise comparisons (e.g., Wilcoxon signed-rank test)
   - Multiple comparison corrections (e.g., Bonferroni)
   - Overall ranking (e.g., Friedman test)

3. **Visualization** (`ranking_plots/`):
   - Box plots comparing metrics across configurations
   - Critical difference diagrams
   - Per-class performance comparison

## Implementation Details

The command should:
- Read `test_history.json` from each configuration's results directory
- Require all configurations to have been evaluated on the same test set
- Support common statistical tests from scipy.stats
- Handle missing metrics gracefully
- Allow customization of:
  - Primary ranking metric
  - Statistical test selection
  - Significance level (α)

## Example Use Case

```bash
# Train three different models
nnBench.train --config unet.yaml --dataset Dataset002_Kits
nnBench.train --config swin_unetr.yaml --dataset Dataset002_Kits  
nnBench.train --config segresnet.yaml --dataset Dataset002_Kits

# Run inference for all
nnBench.inference --config unet.yaml --dataset Dataset002_Kits
nnBench.inference --config swin_unetr.yaml --dataset Dataset002_Kits
nnBench.inference --config segresnet.yaml --dataset Dataset002_Kits

# Compare and rank
nnBench.rank --dataset Dataset002_Kits --configs unet swin_unetr segresnet
```

## Additional Context

This would complement the existing 4-step workflow (plan → train → inference → plot) by adding a 5th step for multi-model comparison.

Reference: Medical imaging challenges often require such ranking methodologies for leaderboard generation and paper submissions.
